packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -282,6 +282,13 @@ export interface FunctionOptions extends EventInvokeConfigOptions {
    * @default false
    */
   readonly allowPublicSubnet?: boolean;
+
+  /**
+   * The ARN of the AWS KMS key that's used to encrypt your function's environment variables.
+   *
+   * @default - AWS Lambda uses a default service key.
+   */
+  readonly kmsKeyArn?: string;
```

Comment:
This needs to be of type [IKey](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-kms.IKey.html).

Several times, I've assumed this to be the kms key for encrypting code and realize it's not until I read the description. We should name this property `environmentEncryption` so as to make it clear.